### PR TITLE
Use `ThreadLocalRandom` in `TrieMap.size`.

### PR DESCRIPTION
**Pull request by axel22**
Review by @jsuereth.

### CODE REVIEW

File: src/library/scala/collection/concurrent/TrieMap.scala [hunk 1]
```diff
@@ -473,7 +473,11 @@ extends CNodeBase[K, V] {
   private def computeSize(ct: TrieMap[K, V]): Int = {
     var i = 0
     var sz = 0
-    val offset = math.abs(util.Random.nextInt()) % array.length
+    val offset =
+      if (array.length > 0)
+        //util.Random.nextInt(array.length) /* <-- benchmarks show that this causes observable contention */
+        scala.concurrent.forkjoin.ThreadLocalRandom.current.nextInt(0, array.length)
```
Comment (jsuereth) [inline src/library/scala/collection/concurrent/TrieMap.scala:479]: Nice find!  What kind of random numbers are we getting here?  I'm guessing not from `/dev/random`.   Just curious for my own usages :)

**Comment by jsuereth** [comment]
LGTM

File: src/library/scala/collection/concurrent/TrieMap.scala (hunk 1 above)
Comment (axel22) [inline src/library/scala/collection/concurrent/TrieMap.scala:479]: No, it's just a pseudorandom generator like a normal Java `Random`, but a separate instance is stored for each thread in the thread local.
When using a global one, writes to it end up causing a lot of contention.

Comment (jsuereth) [inline src/library/scala/collection/concurrent/TrieMap.scala:479]: Yeah....   I know using SecureRandom kills your performance.  I also know
people (monte-carlo simulators) spend lots of time working for
pseudo-random number generates with good performance.   I was only asking
to see how re-usable that particular random number generator will be in my
projects/experiments.

Thanks!

**Pull request merged**